src/foundry/client/data/documents/chat-message.d.ts

```diff
@@ -207,35 +180,45 @@ declare global {
 
     /**
      * Render the inner HTML content for ROLL type messages.
-     * @param messageData - The chat message data used to render the message HTML
+     * @param messageData     - The chat message data used to render the message HTML
      * @internal
      */
     protected _renderRollContent(messageData: ChatMessage.MessageData): Promise<void>;
 
+    /** @internal */
     protected override _preCreate(
-      data: ChatMessageDataConstructorData,
+      data: BaseChatMessage.ConstructorData,
       options: DocumentModificationOptions,
-      user: foundry.documents.BaseUser
+      user: BaseUser
     ): Promise<void>;
 
+    /** @internal */
     protected override _onCreate(
-      data: foundry.data.ChatMessageData["_source"],
+      data: BaseChatMessage.ConstructorData,
```

Comment:
If it was `_source` before, I think the better equivalent now would be `BaseChatMessage.Source`.

Reply:
Updated